### PR TITLE
Simple wallet transaction processor in Golang

### PR DESCRIPTION
**Pull request by arsulegai**
Implement the simple transaction processor in Go language.

Signed-off-by: S m, Aruna <aruna.s.m@intel.com>

### CODE REVIEW

**Comment by arsulegai** [comment]
TODO items are added, except the license header part. Tested it to be working in container environment.

File: goprocessor/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+WORKDIR /go/src/github.com/hyperledger/sawtooth-sdk-go
+RUN go generate 
+
+EXPOSE 4004/tcp
+
+WORKDIR /project/simplewallet/goprocessor
+COPY . ./
+RUN bash -c "./build.sh"
+RUN rm -rf src/ *.sh Dockerfile
+CMD bash -C "simple_wallet"
```
Comment (askmish) [inline goprocessor/Dockerfile]: please keep simple_wallet as a single word "simplewallet" to maintain consistency with TPs in other languages.

File: goprocessor/build.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+go build src/sawtooth_siimplewallet/simple_wallet.go
```
Comment (askmish) [inline goprocessor/build.sh]: please keep "simple_wallet" as a single word "simplewallet" to maintain consistency with TPs in other languages.

File: goprocessor/src/sawtooth_siimplewallet/constants/globals.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package constants
```
Comment (askmish) [inline goprocessor/src/sawtooth_siimplewallet/constants/globals.go]: Copyright headers missing.

File: goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_handler.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
+package handlers
```
Comment (askmish) [inline goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_handler.go]: copyright missing

File: goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_struct.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package handlers
```
Comment (askmish) [inline goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_struct.go]: Typo error in the directory name? goprocessor/src/**sawtooth_siimplewallet**/handlers/simple_wallet_struct.go

**Review by askmish — CHANGES_REQUESTED** [review]
Please address the preliminary review comments across the PR.

File: goprocessor/Dockerfile (hunk 1 above)
Comment (arsulegai) [reply goprocessor/Dockerfile]: Ok

File: goprocessor/build.sh (hunk 2 above)
Comment (arsulegai) [reply goprocessor/build.sh]: Ok

File: goprocessor/src/sawtooth_siimplewallet/constants/globals.go (hunk 3 above)
Comment (arsulegai) [reply goprocessor/src/sawtooth_siimplewallet/constants/globals.go]: Yes, I will put copyright headers as in other files for all files uploaded

File: goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_handler.go (hunk 4 above)
Comment (arsulegai) [reply goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_handler.go]: Yes, I will put copyright headers as in other files for all files uploaded

File: goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_struct.go (hunk 5 above)
Comment (arsulegai) [reply goprocessor/src/sawtooth_siimplewallet/handlers/simple_wallet_struct.go]: Right! I created folder and copy pasted the path across the project, will correct it.

**Comment by arsulegai** [comment]
Added missing copyright headers to all introduced files.

**Comment by danintel** [comment]
I will review this tomorrow, 9/27. In general, it's a great addition to have another language example, Go.

File: goprocessor/src/sawtooth_simplewallet/handlers/simplewallet_handler.go [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+	walletKey := self.getNamespaceAddress() + utils.Hexdigest(self.getUserFrom())[:constants.TransactionUserAddressLength]
+	logger.Info("Got user key " + self.getUserFrom() + "wallet key " + walletKey)
+	// Get balance from ledger state
+	currentLedgerEntry, err := self.getContext().GetState([]string{walletKey})
+	if err != nil {
+		return err
+	}
+	balance := string(currentLedgerEntry[walletKey])
+	newBalance := 0
+	// getState() will return empty map if wallet key doesn't exist in state
```
Comment (danintel) [inline goprocessor/src/sawtooth_simplewallet/handlers/simplewallet_handler.go]: s/getState/GetState/

**Review by danintel — CHANGES_REQUESTED** [review]
One trivial typo, otherwise looks good. Thanks for adding this.

File: goprocessor/src/sawtooth_simplewallet/handlers/simplewallet_handler.go (hunk 6 above)
Comment (arsulegai) [reply goprocessor/src/sawtooth_simplewallet/handlers/simplewallet_handler.go]: Good catch, I will update the commented line

**Comment by arsulegai** [comment]
Updated comment section of code to reflect GetState() instead of getState()

**Review by danintel — APPROVED** [review]

**Review by askmish — APPROVED** [review]

**Pull request merged**